5.5/run-mysqld-slave.sh

```diff
@@ -6,6 +6,7 @@ source ${HOME}/common.sh
 set -eu
 
 export MYSQL_RUNNING_AS_SLAVE=1
+export MYSQL_MASTER_ADDR=${MYSQL_MASTER_ADDR:-mysql-master}
```

Comment:
what provides the MYSQL_MASTER_ADDR env variable?

and are you saying "mysql-master" resolves to something via DNS?  what domain is it searching?